lib/entities/team-membership.js

```diff
@@ -0,0 +1,137 @@
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getOrganizations()
+     * .then((organizations) => organizations.items[0])
+     * .then(org => org.getTeamMembershipByTeam('teamId', 'teamMembership_id'))
+     * .then((teamMembership) => {
+     *  teamMembership.admin = true;
+     *  teamMembership.update();
+     * })
+     * .catch(console.error)
+     */
+
+    update: function () {
+      const raw = this.toPlainObject()
+
+      return http.put('teams/' + teamId + '/team_memberships/' + this.sys.id, raw, {
+        headers: {
+          'X-Contentful-Version': this.sys.version || 0
+        }
+      })
+        .then((response) => wrapTeamMembership(http, response.data), errorHandler)
+    },
+
+    /**
+     * Deletes this object on the server.
+     * @memberof TeamMembership
+     * @func delete
+     * @return {Promise} Promise for the deletion. It contains no data, but the Promise error case should be handled.
+     * @example
+     * const contentful = require('contentful-management')
+     *
+     * const client = contentful.createClient({
+     *   accessToken: '<content_management_api_key>'
+     * })
+     *
+     * client.getOrganizations()
+     * .then((organizations) => organizations.items[0])
+     * .then(org => org.getTeamMembershipByTeam('teamId', 'teamMembership_id'))
+     * .then((teamMembership) => {
+     *  teamMembership.delete();
+     * })
+     * .catch(console.error)
+     */
+
+    delete: function () {
+      return http.delete('teams/' + teamId + '/team_memberships/' + this.sys.id)
+        .then((response) => {}, errorHandler)
+    }
+  }
+}
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw team membership data
+ * @return {TeamMembership} Wrapped team membership data
+ */
+export function wrapTeamMembershipByTeam (http, data, teamId) {
+  const teamMembership = toPlainObject(cloneDeep(data))
+  enhanceWithMethods(teamMembership, createTeamMembershipApi(http, teamId))
+  return freezeSys(teamMembership)
+}
+
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw team membership collection data
+ * @return {TeamMembershipCollection} Wrapped team membership collection data
+ */
+export function wrapTeamMembershipCollectionByTeam (http, data, teamId) {
+  const teamMemberships = toPlainObject(cloneDeep(data))
+  teamMemberships.items = teamMemberships.items.map((entity) => wrapTeamMembershipByTeam(http, entity, teamId))
+  return freezeSys(teamMemberships)
+}
+
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw team membership data
+ * @return {TeamMembership} Wrapped team membership data
+ */
+export function wrapTeamMembership (http, data) {
```

Comment:
I don't see why we have "teamMembership" entity, but then we also have ways to wrap `teamMembershipByTeam` and `teamMembership`. 

Isn't the entity that's being returned always a `teamMembership`? And shouldn't it always have update and delete methods on it?

I think these wrap methods can be consolidated to just `wrapTeamMembership` and `wrapTeamMembershipCollection`, and the former will wrap the entity and create the teamMembershipApi (with update/delete)

Reply:
Yes, I know its a bit confusing, as I mentioned before I needed the wrapper methods because we will also have the `/organization/orgId/team_memberships` and for that we do not want the update and delete.

Reply:
Sorry should have also specified that in the PR description. I stared with the org team memberships but then priorities shifted.